### PR TITLE
[DD-1450] Yearn V1 Integration. None-upgradeable version [WIP]

### PR DESCRIPTION
**Pull request by AlexanderMazaletskiy**

### CODE REVIEW

**Review by Pavel-Blaize — APPROVED** [review]

File: contracts/yearnV1/vault-savings/VaultSavings.sol [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+        baseAmount = IERC20(baseToken).balanceOf(address(this));
+
+        //Transfer token to user
+        IERC20(baseToken).safeTransfer(msg.sender, baseAmount);
+
+        emit Withdraw(_vault, msg.sender, baseAmount, _amount);
+    }
+
+    function registerVault(address _vault) external override {
```
Comment (pash7ka) [inline contracts/yearnV1/vault-savings/VaultSavings.sol:111]: Maybe we can restrict who can call this to `onlyOwner`?

File: contracts/yearnV1/vault-savings/VaultSavings.sol [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            isActive: true,
+            blockNumber: block.number
+        });
+
+        (, address baseToken,  ,  ,) = IYRegistry(registry).getVaultInfo(_vault);
+
+        emit VaultRegistered(_vault, baseToken);
+    }
+
+    function activateVault(address _vault) external override {
```
Comment (pash7ka) [inline contracts/yearnV1/vault-savings/VaultSavings.sol:126]: Same as above, probably has to be restricted to `onlyOwner`

File: contracts/yearnV1/vault-savings/VaultSavings.sol [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        vaults[_vault] = VaultInfo({
+            isActive: true,
+            blockNumber: block.number
+        });
+
+       emit VaultActivated(_vault);
+
+    }
+
+    function deactivateVault(address _vault) external override {
```
Comment (pash7ka) [inline contracts/yearnV1/vault-savings/VaultSavings.sol:138]: Same as above, probably has to be restricted to `onlyOwner`

File: contracts/yearnV1/vault-savings/VaultSavings.sol (hunk 1 above)
Comment (pash7ka) [inline contracts/yearnV1/vault-savings/VaultSavings.sol:111]: Maybe we can restrict who can call this to `onlyOwner`?

**Review by pash7ka — CHANGES_REQUESTED** [review]
Admin functions in VaultSavings need access restrictions.
Otherwise i can see no problems with this.

File: contracts/yearnV1/vault-savings/VaultSavings.sol (hunk 1 above)
Comment (AlexanderMazaletskiy) [reply contracts/yearnV1/vault-savings/VaultSavings.sol:111]: Agree

**Review by pash7ka — APPROVED** [review]

**Pull request merged**